builtins/amp-video.js

```diff
@@ -36,7 +36,31 @@ export function installVideo(win) {
     }
 
     /** @override */
+    prerenderAllowed() {
+      return true;
+    }
+
+    /** @override */
+    buildCallback() {
+      /** @private */
+      this.realPreloadAttrValue_ = this.element.getAttribute('preload');
+      const posterAttr = this.element.getAttribute('poster');
+
+      if (!posterAttr) {
+        console/*OK*/.warn(
+            'No "poster" attribute has been provided for amp-video.');
+      }
+    }
+
+    /** @override */
     layoutCallback() {
+      const docState = documentStateFor(this.getWin());
+      // Disable video preload in prerender mode and reset it when rendered.
+      if (docState.getVisibilityState() === 'prerender') {
```

Comment:
Ah. Got it. Makes since. Is there a state where we know the viewer is fully visible and we can reset the preload attribute on videos? I agree that that resetting the preload at that point might be a better approach.